tests/test_gemini/update/test_optim.py

```diff
@@ -57,8 +58,12 @@ def exam_model_step(placement_policy, model_name: str):
     torch_model, torch_optim = convert_to_apex_amp(torch_model, torch_optim, amp_config)
     torch_model = DDP(torch_model, device_ids=[dist.get_rank()])
 
-    with ColoInitContext(device=get_current_device()):
+    init_dev = get_current_device()
+    with ColoInitContext(device=init_dev):
         model = model_builder()
+
+    post_process_colo_init_ctx(model, device=init_dev)
```

Comment:
Could we add this function when exiting the ColoInitContext?

Reply:
No.... ColoInitContext dose not know the root module.

Reply:
I add more design details in PR description.